src/AfterLogin/SearchPage.js

```diff
@@ -0,0 +1,247 @@
+            <Typography sx={{ width: "5%" }}>
+              {deck.deckCards.length}
+            </Typography>
+          </AccordionSummary>
+          {cardList}
+        </Accordion>
+      );
+    });
+
+  const resultList =
+    result &&
+    result.map((deck) => {
+      const cardListBody = deck.cardInfos.map((card) => {
+        return (
+          <TableRow key={`card${card.cardID}`}>
+            <TableCell>{card.english}</TableCell>
+            <TableCell>{card.spanish}</TableCell>
+          </TableRow>
+        );
+      });
+      const cardList = (
+        <AccordionDetails>
+          <TableContainer component={Paper}>
+            <Table>
+              <TableHead>
+                <TableRow>
+                  <TableCell>
+                    <b>English</b>
+                  </TableCell>
+                  <TableCell>
+                    <b>Spanish</b>
+                  </TableCell>
+                </TableRow>
+              </TableHead>
+              <TableBody>{cardListBody}</TableBody>
+            </Table>
+          </TableContainer>
+          <div>
+            <Button
+              variant="contained"
+              className="copy-button"
+              onClick={() => handleCopy(deck.deckID)}
+            >
+              <div>Copy Deck</div>
+              <ContentCopyIcon />
+            </Button>
+          </div>
+        </AccordionDetails>
+      );
+      return (
+        <Accordion
+          className="search-list"
+          expanded={expanded === deck.deckID}
+          key={`deck${deck.deckID}`}
+          onChange={() => handleExpanded(deck.deckID)}
+        >
+          <AccordionSummary>
+            <Typography sx={{ width: "30%" }}>Deck Name:</Typography>
+            <Typography sx={{ width: "35%" }}>
+              <b>{deck.deckName}</b>
+            </Typography>
+            <Typography sx={{ width: "30%" }}>Deck Cards: </Typography>
+            <Typography sx={{ width: "5%" }}>
+              {deck.deckCards.length}
+            </Typography>
+          </AccordionSummary>
+          {cardList}
+        </Accordion>
+      );
+    });
+
+  const resultDisplay = result && !!result.length && <div>{resultList}</div>;
+
+  const adviceDisplay = advice && !!advice.length && (
+    <div>
+      <div>You may also interested in:</div>
+      <div>{adviceList}</div>
+    </div>
+  );
+
+  const displayList = (
+    <div>
+      {resultDisplay}
+      <br />
+      {adviceDisplay}
+    </div>
+  );
+
+  const display = result ? (
+    displayList
+  ) : (
+    <Backdrop open={!result}>
+      <h3>Searching Deck</h3>
+      <h1>
+        <CircularProgress color="inherit" />
+      </h1>
+    </Backdrop>
+  );
+
+  const pageHeader = (
+    <div className="search-header">
+      {result && result.length
+        ? `${result.length} results of keyword "${keyword}" found:`
+        : `No result of "${keyword}"`}
+    </div>
+  );
```

Comment:
I feel you are overutilizing this pattern of defining smaller stateless components within a stateful component. You should try to abstract more if possible